binance-pipeline/lambdas/binance_historical_extraction.py

```diff
@@ -0,0 +1,249 @@
+# ./lambdas/binance_historical_extraction.py
+from datetime import datetime
+import json
+import math
+import time
+#
+from binance import BinanceQueryAPI
+from binance import TradeIDFinder, get_latest_trade_id, get_first_trade_id
+
+CHUNK_LIMIT = 1e3
+DATEFORMAT = "%Y-%m-%d"
+RATE_LIMIT: int = 12e3
+RESERVED_CONCURRENCY = 4
+SQS_QUEUE_NAME_EXTRACTION = ''
+SQS_QUEUE_NAME_DELAY_DAY = ''
+SQS_QUEUE_NAME_DELAY_HOUR = ''
+SQS_QUEUE_NAME_DELAY_MINUTE = ''
+SQS_QUEUE_NAME_DELAY_SECOND = ''
+
+def parse_date(date: str) -> datetime:
+
+    return datetime.strptime(date, DATEFORMAT)
+
+
+def parse_event(payload: dict):
+    if (type_ := payload['type']) == 'orchestration':
+        # user sent configuration
+        handle_orchestration(payload['config'])
+    elif type_ == 'api_call':
+        # sqs sent the payload to send as GET to API
+        handle_api_call(payload['config'])
+    else:
+        raise Exception("lambda only accepts api_call or orchestration")
+
+    return
+
+
+def ensure_sqs_trigger(record):
+    if (event_source := record["eventSource"]) != "aws:sqs":
+        raise Exception(f"Event source must be aws:sqs not {event_source = }")
+
+    return
+
+
+def handle_orchestration(config: dict):
+    start_date: str = config.get('start_date')
+    end_date: str = config.get('end_date')
+    reverse: bool = config.get('reverse')
+    symbol: str = config.get('symbol')
+
+    if reverse:
+        handle_reverse_load(symbol, start_date)
+    else:
+        if not start_date:
+            handle_from_start(symbol, end_date)
+        else:
+            handle_from_date(symbol, start_date, end_date)
+
+    return
+
+
+def make_config(start_date: str = '', end_date: str = '', reverse: bool = False):
+    config: dict = {}
+    if start_date:
+        config['start_date'] = start_date
```

Comment:
Might be overkill / limiting, but does it make sense to make `config` a dataclass?

Reply:
It would be substantially more pythonic - and a good shout

the config is passed down the queue for each subsequent invocation, so while we can use the __repr__ of the data class to pass into the queue, when we retrieve the message from the queue it may add a little overhead converting it back into the data class to once again be handled. Readability counts however so I will mock something up and see how it improves matters